kitsune/sumo/utils.py

```diff
@@ -128,7 +130,16 @@ def get_next_url(request):
     else:
         url = request.META.get("HTTP_REFERER")
 
-    if not settings.DEBUG and not is_safe_url(
+    if not url:
+        return None
+
+    # Ensure that the URL is allowed as a redirect.
+    try:
+        HttpResponseRedirect(url)
```

Comment:
I am not sure that I completely understand this solution. Can you maybe elaborate on why this is working?

Reply:
I am trying to follow the code. This ends up being something like `abc:abc` which the urlparse does not recognize as a valid scheme (abc) when initializes the HttResponseRedirect

Reply:
I had to dive into the code to start understanding why this is a solution to the problem. Personally I find this a bit cryptic on why it's happening. The short version is:

- request object from HttpRequest reconstruct the query_string by using the `parse_qsl` from urllib.parse
- HttpResposneRedirect constrcuctor is using the HttpResponseRedirectBase where:
  - adds the location header by using the iri_to_uri
  - uses the urlparse from urllib on the passed url from the request which returns a schme of `javascript` in our example
  
Personally I feel it would be easier to understand this code if we had something similar to:
```python
parsed_url = urlparse(url)
if parsed_url.scheme and parsed_url.scheme not in ['http', 'https']:
    return None

if not settings.DEBUG and not url_has_allowed_host_and_scheme(
        url, allowed_hosts={Site.objects.get_current().domain}
    ):
        return None
return iri_to_uri(url)

Reply:
@akatsoulas First of all, thanks for digging into this in more detail! Here are my thoughts: